pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -63,48 +134,52 @@ func (r *ApicastReconciler) apicast() (*component.Apicast, error) {
 	return component.NewApicast(opts), nil
 }
 
-func (r *ApicastReconciler) reconcileDeploymentConfig(desiredDeploymentConfig *appsv1.DeploymentConfig) error {
-	err := r.InitializeAsAPIManagerObject(desiredDeploymentConfig)
+func (r *ApicastReconciler) reconcileStagingDeploymentConfig(desiredDeploymentConfig *appsv1.DeploymentConfig) error {
+	reconciler := NewDeploymentConfigBaseReconciler(r.BaseAPIManagerLogicReconciler, NewApicastStagingDCReconciler(r.BaseAPIManagerLogicReconciler))
+	err := reconciler.Reconcile(desiredDeploymentConfig)
 	if err != nil {
 		return err
 	}
-
-	return r.deploymentConfigReconciler.Reconcile(desiredDeploymentConfig)
+	r.Logger().Info(fmt.Sprintf("%s reconciled", ObjectInfo(desiredDeploymentConfig)))
+	return nil
 }
 
-func (r *ApicastReconciler) reconcileConfigMap(desiredConfigMap *v1.ConfigMap) error {
-	err := r.InitializeAsAPIManagerObject(desiredConfigMap)
+func (r *ApicastReconciler) reconcileProductionDeploymentConfig(desiredDeploymentConfig *appsv1.DeploymentConfig) error {
+	reconciler := NewDeploymentConfigBaseReconciler(r.BaseAPIManagerLogicReconciler, NewApicastStagingDCReconciler(r.BaseAPIManagerLogicReconciler))
```

Comment:
Is NewApicastStagingDCReconciler correct in the case of the reconcileProductionDeploymentConfig ? Shouldn't it be NewApicastProductionDCReconciler ?
In case they share the same behaviour we could name it NewApicastDCReconciler. In case you don't like that because they might differ in the future I'd create another type and make both call the same function and if they differ in the future act accordingly

Reply:
I will change NewApicastDCReconciler and when there are differences, make it explicit.

Reply:
:+1: